Project/Source/Components/ComponentScript.cpp

```diff
@@ -24,8 +24,34 @@
 #define JSON_TAG_TYPE "Type"
 #define JSON_TAG_VALUE "Value"
 
+ComponentScript::~ComponentScript() {
+	App->resources->DecreaseReferenceCount(scriptId);
+
+	for (auto& entry : changedValues) {
+		if (entry.second.first == MemberType::PREFAB_RESOURCE_UID) {
+			UID prefabId = std::get<UID>(entry.second.second);
+			App->resources->DecreaseReferenceCount(prefabId);
+		}
+	}
+}
+
+void ComponentScript::Init() {
+	App->resources->IncreaseReferenceCount(scriptId);
+
+	for (auto& entry : changedValues) {
+		if (entry.second.first == MemberType::PREFAB_RESOURCE_UID) {
+			UID prefabId = std::get<UID>(entry.second.second);
+			App->resources->IncreaseReferenceCount(prefabId);
+		}
+	}
+
+	if (App->project->IsGameLoaded()) {
+		CreateScriptInstance();
+	}
+}
+
 void ComponentScript::Start() {
-	if (scriptInstance && App->time->HasGameStarted() && App->scene->scene->sceneLoaded) {
```

Comment:
Why have these lines been deleted? What if a Script needs the reference of another go that hasn't yet been initialized? Are you calling the Script->Start() from somewhere else?

Reply:
Start only gets called when this condition is true, so it was redundant.